php-checks/src/main/java/org/sonar/php/checks/LocalVariableShadowsClassFieldCheck.java

```diff
@@ -166,10 +188,20 @@ private void checkParameters(AstNode functionDec) {
     }
   }
 
-  private void reportIssue(AstNode node, String varName) {
-    getContext().createLineViolation(this, "Rename \"{0}\" which has the same name as the field declared at line {1}.",
-      node, varName, classState.getLineOfFieldNamed(varName));
-    classState.setAsCheckedVariable(varName);
+  private boolean isExcluded(MethodDeclarationTree methodDec) {
+    String methodName = methodDec.name().text();
+    return isStaticClassMember(methodDec.modifiers())
+      || isConstructor(methodName) || isSetter(methodName);
+  }
+
+  // fixme (Lena) : refactor with using CheckUtils after merge of PR https://github.com/SonarCommunity/sonar-php/pull/43
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)